### PR TITLE
Spec: Apply changes to weak conformance and introduce harmonization.

### PR DESCRIPTION
**Pull request by sjrd**
[skip ci]

Note that weak conformance was not entirely dropped. It is still applicable to determine *compatibility* of a method call argument to the parameter type. See #7708.

@odersky Since you asked about it in the big type system PR. ;) FYI the changes in `TODOreference` and `APPLIEDreference` are part of [our process to integrate the Reference into the Spec](https://github.com/lampepfl/dotty/pull/16839).

### CODE REVIEW

File: docs/_spec/TODOreference/dropped-features/weak-conformance-spec.md [hunk 1]
```diff
@@ -1,54 +0,0 @@
----
-layout: doc-page
```
Comment (odersky) [inline docs/_spec/TODOreference/dropped-features/weak-conformance-spec.md:2]: I think it's valuable to have precise rules in these pages or at least refer to them. Of course, many pages here do not have precise rules, but I'd like us to move in the general direction of more precise rules, not loss.

So I'd either keep this page, or link to the section on Harmonization in the spec, if that's feasible.

**Review by odersky — COMMENTED** [review]
Otherwise LGTM

File: docs/_spec/TODOreference/dropped-features/weak-conformance-spec.md (hunk 1 above)
Comment (sjrd) [reply docs/_spec/TODOreference/dropped-features/weak-conformance-spec.md:2]: We don't publish the spec yet, so I cannot link to it. But I re-introduced an explicit reference to the relevant sections of the spec. (note that the content of `APPLIEDreference` is not what gets published at https://docs.scala-lang.org/scala3/reference/ at the moment, so this does not change anything to the currently published documents).

**Comment by sjrd** [comment]
Ping @odersky. Is this OK to merge?

**Comment by odersky** [comment]
Yes, ready to merge.

**Review by odersky — APPROVED** [review]

**Pull request merged**